### ISSUE TITLE
[Bluesky Web App] Posting links highlights wrong characters/non link text as a link

### ISSUE THREAD

**Issue by Hydroponic-Gardener** [bluesky-social/social-app#1391]
While typing out a post, if I post a link (example, google.com), the link highlighting bleeds out into other text. This isn't shown in the actual post, only while typing, or sometimes doesn't show up as a link. Examples below:

![image](https://github.com/bluesky-social/social-app/assets/130553435/bc16b86b-c663-4610-a016-ee93e60bfbef)

![image](https://github.com/bluesky-social/social-app/assets/130553435/22aeaf88-400e-49d7-a786-3f042af7787b)

To reproduce:

For non-highlighted link:
1) Write out text
2) Paste the link, embed or not
3) Write something before the link, and don't put a space at the end.

Result:

The link will be registered as normal text. 

For highlighted link bleed:
1) Write out text
2) Past the link, embed or not
3) Write text above/below the link, and leave a space before the end of sentence/text.

Result:

The link, depending on how many times one presses enter above the link, will lose the blue highlight, and transfer it on text above/below.

Platform:
Linux (Arch, 6.4.12-zen kernel)
Browser: Firefox 117.0 64-bit.

**Comment by mschwendener** [comment]
Related: https://github.com/bluesky-social/social-app/issues/1353

**Comment by Hydroponic-Gardener** [comment]
> Related: #1353

Didn't see it when searching, I'll remove this post then. Thanks on letting me know!

**Comment by gani1000** [comment]
Hi, can I work on this? I'd like to fix this issue.

**Comment by pfrazee** [comment]
Feel free to submit a PR!

**Comment by andreslucero** [comment]
@gani1000 @pfrazee this may also be related to the encoding of newline characters.

In the first example, it appears the newline isn’t recognized and so the highlight never triggers.

In the second example, the newlines in the middle of the post are ignored, so the highlight starts where the URL *should be* if the newlines didn’t exist.

**Comment by estrattonbailey** [comment]
@Hydroponic-Gardener I commented on the other issue, but same thing here: I can't reproduce this 🤔

I know I've seen this kinda thing in the past but perhaps we bumped something and resolved it? Are you able to reproduce this?

**Comment by mschwendener** [comment]
> In the first example, it appears the newline isn’t recognized and so the highlight 
> never triggers.

Yes, that also happened to me at first now that I wanted to test if it was solved: at first the URLs were just black text. 

But then after adding some other URLs (because I wanted to make a screenshot), the original issue was back. 

<img width="590" alt="screenshot_4567" src="https://github.com/bluesky-social/social-app/assets/4497705/ec881fe7-f217-4574-a9fe-92bd86255bac">

**Comment by estrattonbailey** [comment]
Oooh ok, saw in the other ticket that you were using `Shift + Return`. Apologies, wasn't thinking about the significance of "carriage return" :P 

So yep, this is sorta related to character encoding, and it should be fixed by https://github.com/bluesky-social/social-app/pull/1544, which will go out in the next release. Tested these cases locally with that branch and it appears to work fine.

I'll keep this open until then though and we can confirm before closing 👍 thank you all for your help! Appreciate your patience.

**Comment by mschwendener** [comment]
> wasn't thinking about the significance of "carriage return" :P

am old enough to know what "carriage return" refers to  ;-)   

 
(also, English is not my first language)

**Comment by arturo32** [comment]
@estrattonbailey should this be closed? It doesn't seem to be reproducible anymore.

**Comment by theohonohan** [comment]
(I'm new here.) 

I've just seen a similar problem in the web app. The post is https://bsky.app/profile/nyreviewofarch.bsky.social/post/3lxcy7b4f4c2s

I hope this screenshot captures enough of what's going on. There's a string of three replacement characters in the post text very close to where the problem occurs.

<img width="1429" height="451" alt="Image" src="https://github.com/user-attachments/assets/5c538529-52a9-402c-9a8b-89edbeefac70" />
